lib/executor/constant_arrival_rate.go

```diff
@@ -135,7 +135,7 @@ func (carc ConstantArrivalRateConfig) Validate() []error {
 	}
 
 	if !carc.MaxVUs.Valid {
-		errors = append(errors, fmt.Errorf("the number of maxVUs isn't specified"))
+		carc.MaxVUs.Int64 = carc.PreAllocatedVUs.Int64
```

Comment:
Hmm this seems like a bit of a hack to me... I'd expect `Validate()` to not make any modifications, but yeah, unfortunately there's no other way to do it currently, without significant refactoring... :disappointed: That refactoring is probably going to have to be done as a part of https://github.com/loadimpact/k6/issues/1499 / https://github.com/loadimpact/k6/issues/1427. Until then, can you please add some code comments here, and also add a not in https://github.com/loadimpact/k6/issues/1427 to fix this code as well?

Reply:
fixed in  43ab37d

Reply:
leaving this unresolved, so it's even more apparent that it needs to be fixed